### PR TITLE
type metadata helper refactor

### PR DESCRIPTION
**Pull request by tomnelson**

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/util/MetadataHelperCacheManagementListener.java [hunk 1]
```diff
@@ -24,16 +24,16 @@
     private static final Logger log = Logger.getLogger(MetadataHelperCacheManagementListener.class);
     
     private String zookeepers;
-    private MetadataHelper metadataHelper;
+    private TypeMetadataHelper typeMetadataHelper;
```
Comment (brianloss) [inline warehouse/query-core/src/main/java/datawave/query/util/MetadataHelperCacheManagementListener.java]: Pardon my ignorance here, but it appears to me that MetadataHelper and TypeMetadataHelper are entirely different classes. Why this change (and the same change in the MetadatHelperUpdateHdfsListener class)?

File: warehouse/query-core/src/main/java/datawave/query/util/AllFieldMetadataHelper.java [hunk 2]
```diff
@@ -507,7 +507,7 @@ public CompositeMetadata getCompositeMetadata(Set<String> ingestTypeFilter) thro
     @Cacheable(value = "getFieldsForDatatype", key = "{#datawaveType,#ingestTypeFilter}", cacheManager = "metadataHelperCacheManager")
```
[the remark this answers is not in the document]
Comment (brianloss) [reply warehouse/query-core/src/main/java/datawave/query/util/AllFieldMetadataHelper.java]: I think you could add `#root.target.metadataTableName` to the key to cover this, unless metadataTableName needs to be pushed into the method signature...

[the remark this answers is not in the document]
Comment (brianloss) [reply warehouse/query-core/src/main/java/datawave/query/util/AllFieldMetadataHelper.java]: You didn't change it in this PR, but there's also a caching problem on the method `public Set<String> getFieldsForDatatype(Class<? extends Type<?>> datawaveType)`. Internally, it calls this method. However, since it's an internal call, the caching annotation on this method won't be honored. Therefore, if you modify the cache key on this method to include the metadataTableName variable, you must also modify the cache key on the ``public Set<String> getFieldsForDatatype(Class<? extends Type<?>> datawaveType)` variant.

[the remark this answers is not in the document]
Comment (tomnelson) [reply warehouse/query-core/src/main/java/datawave/query/util/AllFieldMetadataHelper.java]: I don't believe that the metadataTableName will ever change in the MetadataHelper or in the AllFieldMetadataHelper, once the instance is initialized.

File: warehouse/query-core/src/main/java/datawave/query/util/MetadataHelperCacheManagementListener.java (hunk 1 above)
Comment (tomnelson) [reply warehouse/query-core/src/main/java/datawave/query/util/MetadataHelperCacheManagementListener.java]: What goes into hdfs comes from the TypeMetadatahelper. The MetadataHelper has a TypeMetadataHelper, but this change it to make the connection more direct. This way, the code that uses the TypeMetadataHelper will not be encumbered by the other baggage/dependencies that come with the MetadataHelper instance tree.
The metadataTableName is no longer stateful in the TypeMetadataHelper. One instance of the TypeMetadataHelper may be used to update the hdfs cache of all of the metadata tables.
The metadataTableName remains stateful in the MetadataHelper, because the MetadataHelper is created and initialized for a specific metadata table, and that should not change.
